project/src/Luna/Project.hs

```diff
@@ -22,89 +25,113 @@ import qualified System.FilePath.Find   as Find
 
 lunaFileExt :: String
 lunaFileExt = ".luna"
 
-mkQualName :: Path.Path Path.Rel Path.File -> QualName
-mkQualName file = qualName
+mkQualName :: Name -> Path Rel File -> QualName
+mkQualName projectName file = qualName
     where
-        qualName        = Qual.mkQualName (convert path) (convert moduleName)
+        qualName        = Qual.mkQualName (convert (convert projectName : path)) (convert moduleName)
         path            = filter (/= ".") $ FP.splitDirectories dir
         moduleName      = FP.dropExtensions filename
         (dir, filename) = FP.splitFileName (Path.toFilePath file)
 
-assignQualName :: Path.Path Path.Abs Path.Dir -> Path.Path Path.Abs Path.File -> (Path.Path Path.Abs Path.File, QualName)
-assignQualName srcDir filePath = (filePath, qualName)
+assignQualName :: Path Abs Dir -> Path Abs File -> (Path Abs File, QualName)
+assignQualName project filePath = (filePath, qualName)
     where
-        qualName         = mkQualName relFileName
-        Just relFileName = Path.stripDir srcDir filePath
+        qualName         = mkQualName (getProjectName project) relFileName
+        Just relFileName = Path.stripDir (project </> sourceDirectory) filePath
 
-sourceDirectory :: Path.Path Path.Rel Path.Dir
+sourceDirectory :: Path Rel Dir
 sourceDirectory = $(Path.mkRelDir "src")
 
-findProjectSources :: Path.Path Path.Abs Path.Dir -> IO (Bimap (Path.Path Path.Abs Path.File) QualName)
+findProjectSources :: Path Abs Dir -> IO (Bimap (Path Abs File) QualName)
 findProjectSources project = do
-    let srcDir            = project Path.</> sourceDirectory
+    let srcDir            = project </> sourceDirectory
         lunaFilePredicate = Find.extension Find.~~? lunaFileExt
     lunaFiles    <- Find.find Find.always lunaFilePredicate (Path.toFilePath srcDir)
     lunaFilesAbs <- mapM Path.parseAbsFile lunaFiles
-    let modules  = map (assignQualName srcDir) lunaFilesAbs
+    let modules  = map (assignQualName project) lunaFilesAbs
     return $ Bimap.fromList modules
 
-localLibsPath :: Path.Path Path.Rel Path.Dir
+localLibsPath :: Path Rel Dir
 localLibsPath = $(Path.mkRelDir "local_libs")
 
-listDependencies :: Path.Path Path.Abs Path.Dir -> IO [(Name, FP.FilePath)]
+listDependencies :: Path Abs Dir -> IO [(Name, FP.FilePath)]
 listDependencies projectSrc = do
-    let lunaModules     = projectSrc Path.</> localLibsPath
+    let lunaModules     = projectSrc </> localLibsPath
         lunaModulesPath = Path.toFilePath lunaModules
     dependencies <- tryAny $ Dir.listDirectory lunaModulesPath
     case dependencies of
         Left exc         -> return []
         Right directDeps -> do
             indirectDeps <- forM directDeps $ \proj -> do
                 path <- Path.parseRelDir proj
-                listDependencies (lunaModules Path.</> path)
+                listDependencies (lunaModules </> path)
             return $ map (convert &&& (lunaModulesPath FP.</>)) directDeps <> concat indirectDeps
```

Comment:
Long line! Try:

```hs
             return $  map (convert &&& (lunaModulesPath FP.</>)) directDeps 
                    <> concat indirectDeps
```

Reply:
I'd prefer :

Reply:
I'd prefer slight change (we use this style everywhere):
```
             return $ map (convert &&& (lunaModulesPath FP.</>)) directDeps 
                   <> concat indirectDeps
```